stdlib/public/core/Stride.swift.gyb

```diff
@@ -125,22 +131,54 @@ public func -= <T : UnsignedInteger>(
 
 //===----------------------------------------------------------------------===//
 
-/// An iterator for the result of `stride(from:to:)`.
+extension Strideable {
+  public static func _step(
+    after current: (index: Int?, value: Self),
+    from start: Self, by distance: Self.Stride
+  ) -> (index: Int?, value: Self) {
+    return (nil, current.value + distance)
+  }
+}
+
+extension Strideable where Stride : FloatingPoint {
+  public static func _step(
+    after current: (index: Int?, value: Self),
+    from start: Self, by distance: Self.Stride
+  ) -> (index: Int?, value: Self) {
+    if let i = current.index {
+      return (i + 1, start + Stride(i + 1) * distance)
+    }
+    // If current.index == nil, either we're just starting out (in which case
+    // the next index is 1), or we should proceed without an index just as
+    // though this floating point specialization doesn't exist.
+    return (current.value == start ? 1 : nil, current.value + distance)
+  }
+}
+
+/// An iterator for `StrideTo<Element>`.
 public struct StrideToIterator<Element : Strideable> : IteratorProtocol {
-  internal var _current: Element
+  internal let _start: Element
   internal let _end: Element
   internal let _stride: Element.Stride
+  internal var _current: (index: Int?, value: Element)
```

Comment:
It’s a nit, but in general it would be better to avoid reordering properties, to minimize diffs.